packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -1143,6 +1143,12 @@ export class DatabaseInstanceReadReplica extends DatabaseInstanceNew implements
   constructor(scope: Construct, id: string, props: DatabaseInstanceReadReplicaProps) {
     super(scope, id, props);
 
+    if (props.sourceDatabaseInstance.engine
+        && !props.sourceDatabaseInstance.engine.supportsReadReplicaBackups
+        && props.backupRetention) {
+      throw new Error(`Cannot set \`backupRetention\` for a ${props.sourceDatabaseInstance.engine.engineType} read replica. Automatic backups are not supported.`);
+    }
```

Comment:
The error message seems a little contrived... how about:

```suggestion
      throw new Error(`Cannot set 'backupRetention', as engine '${engineDescription(engine)}' does not support automatic backups for read replicas`);
```